### DISCUSSION TITLE
Supporting named groups in custom variants

### DISCUSSION THREAD

**Discussion by reidbarber** [tailwindlabs/tailwindcss#12073] [Help] [answered]
Is there a way to support [named groups](https://tailwindcss.com/docs/hover-focus-and-other-states#differentiating-nested-groups) (i.e. nested) when adding [custom variants](https://tailwindcss.com/docs/plugins#adding-variants)?

Context: https://github.com/adobe/react-spectrum/issues/5085 / [Plugin source](https://github.com/adobe/react-spectrum/blob/main/packages/tailwindcss-react-aria-components/src/index.js)

Thanks!

**Comment by wongjn** [comment, marked as answer]
Yes it is possible. You'd want to look at adapting the code in core:

https://github.com/tailwindlabs/tailwindcss/blob/0bd81a06c499be58bf87ca16481333823d86e828/src/corePlugins.js#L152-L202

For a simplified example, here's a custom `group-current` implementation that matches when the `group` element has the `current` class:

```js
require('tailwindcss/plugin')(({ matchVariant }) => {
  matchVariant(
    'group',
    (value= '', { modifier }) =>
      modifier
        ? `:merge(.group\\/${modifier}).current &`
        : `:merge(.group).current &`,
    { values: { 'current': 'current' } },
  );
})
```

[See this in Tailwind Play](https://play.tailwindcss.com/RW1ppVates?file=config)

**Reply by reidbarber** [reply]
Thank you!
